pylint/lint/pylinter.py

```diff
@@ -250,6 +260,71 @@ def _load_reporter_by_class(reporter_class: str) -> type[BaseReporter]:
 }
 
 
+def _handle_force_color_no_color(
```

Comment:
I changed this to not change the passed in `list` but instead just return a new list. This fixed your `mypy` issues as well as being a preferred pattern in general.

I also made it so do we loop over the list once instead of doing so twice

Reply:
> I changed this to not change the passed in `list` but instead just return a new list

I'm happy that `sub_reporters = _handle_force_color_no_color(sub_reporters)` "is a valid solution" in the code 😀 

> I also made it so do we loop over the list once instead of doing so twice

Indeed. But this is one-time setup - therefore "there is ample time to do things" 😅 

I feel that the original code (https://github.com/pylint-dev/pylint/pull/9128/files/4427cf3cdde5d2490a7498a4b9d80dfc96724fa1..51cf8d2fcbfa852449d3208b91eca9bbbbf61729) was a bit more straightforward logic/reading-wise

(I'm fine either way - just saying my view)

Reply:
I personally prefer this pattern and considering you're fine with it, let's keep as is.

Thanks for sharing your opinion though, always appreciate being able to discuss different preferences in these reviews :)